Library/Homebrew/cmd/tests.rb

```diff
@@ -41,9 +41,14 @@ def tests
       args << "--trace" if ARGV.include? "--trace"
       if ARGV.value("only")
         ENV["HOMEBREW_TESTS_ONLY"] = "1"
-        test_name, test_method = ARGV.value("only").split("/", 2)
-        args << "TEST=test_#{test_name}.rb"
-        args << "TESTOPTS=--name=test_#{test_method}" if test_method
+
+        pairs = ARGV.value("only").split
```

Comment:
`ARGV.value` only ever returns a single value. To allow something like `brew tests --only=cmd_audit/audit_desc --only=formulary/to_rack` to work you'd need to declare something like:

```diff
--- a/Library/Homebrew/extend/ARGV.rb
+++ b/Library/Homebrew/extend/ARGV.rb
@@ -107,6 +107,13 @@ module HomebrewArgvExtension
     flag_with_value.strip_prefix(arg_prefix) if flag_with_value
   end
 
+  def values(name)
+    arg_prefix = "--#{name}="
+    flags_with_values = select { |arg| arg.start_with?(arg_prefix) }
+    return [] if flags_with_values.empty?
+    p flags_with_values.map { |arg| arg.strip_prefix(arg_prefix) }
+  end
+
```

and do:

```diff
--- a/Library/Homebrew/cmd/tests.rb
+++ b/Library/Homebrew/cmd/tests.rb
@@ -39,11 +39,9 @@ module Homebrew
 
       args = []
       args << "--trace" if ARGV.include? "--trace"
-      if ARGV.value("only")
+      unless (only = ARGV.values("only")).empty?
         ENV["HOMEBREW_TESTS_ONLY"] = "1"
-
-        pairs = ARGV.value("only").split
-        pairs.each do |pair|
+        only.each do |pair|
           test_name, test_method = pair.split("/", 2)
```

However, even with that the multiple `TEST=` arguments don't seem to do anything 😭